test/extended/openstack/loadbalancers.go

```diff
@@ -141,14 +145,108 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb] The Openstack
+			deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
+				testDeployment, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Creating first Openshift service")
+			svcName1 := "udp-lb-shared1-svc"
+			svcPort1 := int32(8082)
+			jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName1)
+			jig.Labels = labels
+			svc1, err := jig.CreateLoadBalancerService(5*time.Minute, func(svc *v1.Service) {
+				svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: svcPort1, TargetPort: intstr.FromInt(8081)}}
+				svc.Spec.Selector = labels
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			o.Expect(svc1.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]).ShouldNot(o.BeEmpty(),
+				"load-balancer-id annotation missing")
+			loadBalancerId := svc1.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+			e2e.Logf("detected loadbalancer id is %q", loadBalancerId)
+
+			g.By(fmt.Sprintf("Creating second Openshift service using the existing Openstack loadbalancer %s", loadBalancerId))
+			svcName2 := "udp-lb-shared2-svc"
+			svcPort2 := int32(8083)
+			jig = e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName2)
+			jig.Labels = labels
+			svc2, err := jig.CreateLoadBalancerService(5*time.Minute, func(svc *v1.Service) {
+				svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: svcPort2, TargetPort: intstr.FromInt(8081)}}
+				svc.Spec.Selector = labels
+				svc.SetAnnotations(map[string]string{"loadbalancer.openstack.org/load-balancer-id": loadBalancerId})
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Checks from openstack perspective")
+			lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(lb.Pools).Should(o.HaveLen(2), "Unexpected number of pools on Openstack LoadBalancer %q", lb.Name)
+
+			g.By("Connectivity checks")
+			o.Expect(svc1.Status.LoadBalancer.Ingress[0].IP).Should(o.Equal(svc2.Status.LoadBalancer.Ingress[0].IP),
+				"Unexpected external IP for svcs sharing the loadbalancer")
+			fip := svc1.Status.LoadBalancer.Ingress[0].IP
+			podNames, err := exutil.GetPodNamesByFilter(oc.KubeClient().CoreV1().Pods(oc.Namespace()),
+				exutil.ParseLabelsOrDie(""), func(corev1.Pod) bool { return true })
+			o.Expect(err).NotTo(o.HaveOccurred())
+			podName, err := getPodNameThroughUdpLb(fip, fmt.Sprintf("%d", svcPort1), "hostname")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podName).Should(o.BeElementOf(podNames))
+			e2e.Logf("Pod %s successfully accessed through svc loadbalancer on ip %s and port %d", podName, fip, svcPort1)
+			podName, err = getPodNameThroughUdpLb(fip, fmt.Sprintf("%d", svcPort2), "hostname")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podName).Should(o.BeElementOf(podNames))
```

Comment:
the code block from 236-239 looks pretty similar, perhaps a slice could get created with the ports used on the sevices and we could iterator over those. It could also help with lenght of pools as being 2, a len of the list of ports could be used instead.

Reply:
Done